langchain/chat_models/openai.py

```diff
@@ -453,6 +454,22 @@ def _invocation_params(self) -> Mapping[str, Any]:
             openai.proxy = {"http": self.openai_proxy, "https": self.openai_proxy}  # type: ignore[assignment]  # noqa: E501
         return {**openai_creds, **self._default_params}
 
+    def _get_invocation_params(
+        self, stop: Optional[List[str]] = None, **kwargs: Any
+    ) -> Dict[str, Any]:
+        """Get the parameters used to invoke the model FOR THE CALLBACKS."""
+        functions = kwargs.get("functions") or (
+            [format_tool_to_openai_function(tool) for tool in kwargs["tools"]]
```

Comment:
confused by this logic. Why would tools get passed here?

Reply:
Yeah agreed. Was a blind copy

Reply:
Good catch. Was cargo culting the JS logic